lightly_studio/src/lightly_studio/resolvers/dataset_resolver/get_root_datasets_details.py

```diff
@@ -0,0 +1,41 @@
+"""Handler for database operations related to fetching root datasets with details."""
+
+from __future__ import annotations
+
+from sqlmodel import Session, col, func, select
+
+from lightly_studio.models.dataset import DatasetDashboardView, DatasetTable
+from lightly_studio.models.sample import SampleTable
+
+
+def get_root_datasets_details(session: Session) -> list[DatasetDashboardView]:
+    """Get root datasets with detailed metadata including sample counts."""
+    datasets_query = (
+        select(  # type: ignore[call-overload]
+            DatasetTable.dataset_id,
+            DatasetTable.name,
+            DatasetTable.sample_type,
+            DatasetTable.created_at,
+            func.count(SampleTable.dataset_id).label("sample_count"),  # type: ignore[arg-type]
```

Comment:
The type issue should be resolved by casting the field to a column:
```suggestion
            func.count(col(SampleTable.dataset_id)).label("sample_count"),
```